config/vault.go

```diff
@@ -219,6 +231,11 @@ func (c *VaultConfig) Finalize() {
 	}
 	c.SSL.Finalize()
 
+	// Order of precedence
+	// 1. `token` configuration value
+	// 2. `VAULT_TOKEN` environment variable
+	// 3. `~/.vault-token` file
+	// 4. `vault_agent_token_file` configuration value
```

Comment:
I think the order here needs to be 4, 2, 1, 3.

Reply:
Thanks, @jefferai for pointing this out. I update PR accordingly.